tv_apps/fling-player/js/connector.js

```diff
@@ -75,6 +77,24 @@
     mDBG.log('Connector#sendMsg');
     mDBG.log('msg = ', msg);
     this._connection.send(castingMessage.stringify(msg));
+    var txt = castingMessage.stringify(msg);
+    if (txt) {
+      this._connection.send(txt);
+    }
+  };
+
+  proto.getControllingDeviceInfo = function () {
+    var resolve;
+    var p = new Promise((res, rej) => {
+      resolve = res;
+    });
+
+    if (this._controllingDeviceInfo) {
+      resolve(this._controllingDeviceInfo);
+    } else {
+      this._deviceInfoResolvers.push(resolve);
```

Comment:
I think I figured out what you're trying to do but the logic here is just so discrete to understand..